examples/name-service/programs/name-service-without-macros/tests/test.rs

```diff
@@ -0,0 +1,494 @@
+#![cfg(feature = "test-sbf")]
+
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::{AnchorDeserialize, InstructionData, ToAccountMetas};
+use light_client::indexer::test_indexer::TestIndexer;
+use light_client::indexer::{AddressMerkleTreeAccounts, Indexer, StateMerkleTreeAccounts};
+use light_client::rpc::merkle_tree::MerkleTreeExt;
+use light_client::rpc::test_rpc::ProgramTestRpcConnection;
+use light_sdk::account_meta::LightAccountMeta;
+use light_sdk::address::{derive_address, derive_address_seed};
+use light_sdk::compressed_account::CompressedAccountWithMerkleContext;
+use light_sdk::error::LightSdkError;
+use light_sdk::instruction_data::LightInstructionData;
+use light_sdk::merkle_context::{AddressMerkleContext, RemainingAccounts};
+use light_sdk::utils::get_cpi_authority_pda;
+use light_sdk::verify::find_cpi_signer;
+use light_sdk::{PROGRAM_ID_ACCOUNT_COMPRESSION, PROGRAM_ID_LIGHT_SYSTEM, PROGRAM_ID_NOOP};
+use light_test_utils::test_env::{setup_test_programs_with_accounts_v2, EnvAccounts};
+use light_test_utils::{RpcConnection, RpcError};
+use name_service_without_macros::{CustomError, NameRecord, RData};
+use solana_sdk::instruction::{Instruction, InstructionError};
+use solana_sdk::native_token::LAMPORTS_PER_SOL;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::{Keypair, Signer};
+use solana_sdk::transaction::{Transaction, TransactionError};
+
+#[tokio::test]
+async fn test_name_service() {
+    let (mut rpc, env) = setup_test_programs_with_accounts_v2(Some(vec![(
+        String::from("name_service_without_macros"),
+        name_service_without_macros::ID,
+    )]))
+    .await;
+    let payer = rpc.get_payer().insecure_clone();
+
+    let mut test_indexer: TestIndexer<ProgramTestRpcConnection> = TestIndexer::new(
+        &[StateMerkleTreeAccounts {
+            merkle_tree: env.merkle_tree_pubkey,
+            nullifier_queue: env.nullifier_queue_pubkey,
+            cpi_context: env.cpi_context_account_pubkey,
+        }],
+        &[AddressMerkleTreeAccounts {
+            merkle_tree: env.address_merkle_tree_pubkey,
+            queue: env.address_merkle_tree_queue_pubkey,
+        }],
+        true,
+        true,
+    )
+    .await;
+
+    let name = "example.io";
+
+    let mut remaining_accounts = RemainingAccounts::default();
+
+    let address_merkle_context = AddressMerkleContext {
+        address_merkle_tree_pubkey: env.address_merkle_tree_pubkey,
+        address_queue_pubkey: env.address_merkle_tree_queue_pubkey,
+    };
+
+    let address_seed = derive_address_seed(
+        &[b"name-service", name.as_bytes()],
+        &name_service_without_macros::ID,
+    );
+    let address = derive_address(&address_seed, &address_merkle_context);
```

Comment:
can we create derive_address_from_seeds() or something that wraps these two?

Reply:
I changed `derive_address` signature to return a tuple of `(address, address_seed)`. I think that's the most intuitive.

`derive_address`, which returns both address and a seed, is now the only public function meant to be used by developers.

I named the functions used underneath as:

- `derive_address_seed`
- `derive_address_from_seed`

and made them `pub(crate)`, so they can't be used by developers.